sdk/core/core-tracing/src/instrumenter.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  Instrumenter,
+  InstrumenterSpanOptions,
+  TracingContext,
+  TracingSpan,
+  TracingSpanContext,
+} from "./interfaces";
+import { createTracingContext } from "./tracingContext";
+
+export function createDefaultTracingSpan(): TracingSpan {
+  return {
+    end: () => {
+      // noop
+    },
+    spanContext() {
+      // The world could always use more zeroes.
+      return {
+        spanId: "00000000-0000-0000-0000-000000000000",
+        traceId: "00000000-0000-0000-0000-000000000000",
+        traceFlags: 0x0,
+      };
+    },
+    isRecording: () => false,
+    recordException: () => {
+      // noop
+    },
+    setAttribute: () => {
+      // noop
+    },
+    setStatus: () => {
+      // noop
+    },
+  };
+}
+
+export function createDefaultInstrumenter(): Instrumenter {
+  return {
+    createRequestHeaders: (): Record<string, string> => {
+      return {};
+    },
+    parseTraceparentHeader: (): TracingSpanContext | undefined => {
+      return undefined;
+    },
+    startSpan: (
+      _name: string,
+      spanOptions: InstrumenterSpanOptions
+    ): { span: TracingSpan; tracingContext: TracingContext } => {
+      return {
+        span: createDefaultTracingSpan(),
+        tracingContext: createTracingContext({ parentContext: spanOptions.tracingContext }),
+      };
+    },
+    withContext<
+      CallbackArgs extends unknown[],
+      Callback extends (...args: CallbackArgs) => ReturnType<Callback>
+    >(
+      _context: TracingContext,
+      callback: Callback,
+      ...callbackArgs: CallbackArgs
+    ): ReturnType<Callback> {
+      return callback(...callbackArgs);
+    },
+  };
+}
+
+/** @internal */
+let instrumenterImplementation: Instrumenter = createDefaultInstrumenter();
```

Comment:
is it worth lazily initializing this in `getInstrumenter()`?

Reply:
Seems reasonable to me - done!